### PR TITLE
Set up new IAM user and access keys

### PR DESCRIPTION
**Pull request by svevang**
Related to https://github.com/PRX/augury.prx.org/issues/545. Sets up some iam users to access the ad-files bucket and patches them into the augury environment.

### CODE REVIEW

File: stacks/apps/augury.yml [hunk 1]
```diff
@@ -369,6 +399,12 @@ Resources:
               Value: !Ref SecretsVersion
             - Name: AWS_DEFAULT_REGION
               Value: !Ref AWS::Region
+            - Name: UPLOAD_ADFILE_SECRET_ACCESS_KEY
```
Comment (svevang) [inline stacks/apps/augury.yml:402]: Hrm,  now that I think about it, I wonder if these will be patched into the container.  I suppose the container is launched with the addition of these ECS env variables?

Comment (farski) [reply stacks/apps/augury.yml:402]: Yes, those are passed in to ` --env` of `docker run`

File: stacks/apps/augury.yml [hunk 2]
```diff
@@ -300,7 +300,37 @@ Resources:
         - { Key: prx:dev:family, Value: Dovetail }
         - { Key: prx:dev:application, Value: Augury }
       TaskRoleArn: !GetAtt TaskRole.Arn
-
+  UploadHttpSigningUser:
```
Comment (farski) [inline stacks/apps/augury.yml:303]: I would probably change this resource logical ID (and the others) to something like `S3SigningUser` or similar. This name came from when it was part of `upload.prx.org` and part of an HTTP API.

File: stacks/apps/augury.yml [hunk 3]
```diff
@@ -300,7 +300,37 @@ Resources:
...
+    Type: AWS::IAM::User
+    Properties:
+      Tags:
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:cloudformation:root-stack-name, Value: !Ref RootStackName }
+        - { Key: prx:cloudformation:root-stack-id, Value: !Ref RootStackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:application, Value: Augury S3 Uploads }
```
Comment (farski) [inline stacks/apps/augury.yml:313]: The `prx:dev:application` tag here should just be `Augury`.

File: stacks/apps/augury.yml [hunk 4]
```diff
@@ -300,7 +300,37 @@ Resources:
...
+      PolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Action:
+              - s3:AbortMultipartUpload
+              - s3:ListMultipartUploadParts
+              - s3:PutObject
+            Resource:
+              - arn:aws:s3:::prx-mrr-us-east-1-ad-files/upload/*
```
Comment (farski) [inline stacks/apps/augury.yml:330]: Let's add a TODO here to remember to handle this better for multi-region support.

File: stacks/apps/augury.yml [hunk 5]
```diff
@@ -369,6 +399,12 @@ Resources:
               Value: !Ref SecretsVersion
             - Name: AWS_DEFAULT_REGION
               Value: !Ref AWS::Region
+            - Name: UPLOAD_ADFILE_SECRET_ACCESS_KEY
+              Value: !GetAtt UploadHttpSigningAccessKey.SecretAccessKey
+            - Name: UPLOAD_ADFILE_ACCESS_KEY_ID
+              Value: UploadHttpSigningAccessKey
```
Comment (farski) [inline stacks/apps/augury.yml:405]: This probably was meant to be `!Ref UploadHttpSigningAccessKey`

**Review by farski — CHANGES_REQUESTED** [review]

File: stacks/apps/augury.yml [hunk 6]
```diff
@@ -300,7 +300,38 @@ Resources:
...
+    Properties:
+      Tags:
+        - { Key: prx:meta:tagging-version, Value: "2021-04-07" }
+        - { Key: prx:cloudformation:stack-name, Value: !Ref AWS::StackName }
+        - { Key: prx:cloudformation:stack-id, Value: !Ref AWS::StackId }
+        - { Key: prx:cloudformation:root-stack-name, Value: !Ref RootStackName }
+        - { Key: prx:cloudformation:root-stack-id, Value: !Ref RootStackId }
+        - { Key: prx:ops:environment, Value: !Ref EnvironmentType }
+        - { Key: prx:dev:application, Value: Augury }
+  UploadHttpSigningAccessKey:
```
Comment (farski) [inline stacks/apps/augury.yml:314]: I'd rename all the remaining `UploadHttp…` resources to match `S3Signing…`. Other than that I think this looks good.

**Review by farski — CHANGES_REQUESTED** [review]

**Review by farski — APPROVED** [review]

File: stacks/apps/augury.yml [hunk 7]
```diff
@@ -315,7 +316,7 @@ Resources:
     Type: AWS::IAM::AccessKey
     Properties:
       UserName: !Ref S3SigningUser
-  UploadHttpSigningS3AccessPolicy:
+  S3SigningAccessPolicy:
```
Comment (svevang) [inline stacks/apps/augury.yml:319]: Gah, missed this!  Thank you @farski !

**Pull request merged**